.github/workflows/compile.yml

```diff
@@ -0,0 +1,77 @@
+name: Build policies
+on:
+  - pull_request
+  - push
+
+jobs:
+  compile:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        name:
+          - Fedora 33 GCC
+          - Fedora 33 GCC strict
+          - Fedora 33 Clang
+          - Fedora 33 Clang strict
+          - Fedora 33 Clang -Weverything
+
+        include:
+          - name: Fedora 33 GCC
+            image: fedora:33
+            compiler: gcc
+            cpp-compiler: g++
+
+          - name: Fedora 33 GCC strict
+            image: fedora:33
+            compiler: gcc
+            cpp-compiler: g++
+            cflags:  "-O2 -g -Wall -Wextra -Wundef -Wpointer-arith -Wfloat-equal \
+            -Wstrict-prototypes -Wstrict-overflow=5 -Wwrite-strings -Winit-self \
+            -Wuninitialized -Wno-sign-compare -Wshadow -Wformat-security"
+
+          - name: Fedora 33 Clang
+            image: fedora:33
+            compiler: clang
+            cpp-compiler: clang++
+            cflags: "-Wno-error=unused-command-line-argument"
+
+          - name: Fedora 33 Clang strict
+            image: fedora:33
+            compiler: clang
+            cpp-compiler: clang++
+            cflags: "-Wall -Wextra -Wno-error=unused-command-line-argument"
+
+          - name: Fedora 33 Clang -Weverything
+            image: fedora:33
+            compiler: clang
+            cpp-compiler: clang++
+            cflags: "-Weverything -Wno-error=unused-command-line-argument"
+
+    container:
+      image: ${{ matrix.image }}
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install build dependencies
+        run: "dnf -y install 389-ds-base 389-ds-base-snmp autoconf automake \
+bind-utils bzip2 bzip2-devel cargo clang coreutils-common \
+cracklib-devel cracklib-dicts dnf-utils doxygen gcc gcc-c++ \
+git hostname icu krb5-devel libasan libcmocka-devel libdb-devel \
+libevent-devel libicu-devel libtalloc-devel libtevent-devel \
+libtool libubsan lm_sensors-devel make net-snmp-devel net-tools \
+nspr-devel nss-devel openldap-devel openssl openssl-devel pam-devel \
+pcre-devel procps-ng psmisc python3-argcomplete python3-argparse-manpage \
+python3-devel python3-ldap python3-libselinux python3-pip \
+python3-policycoreutils python3-pyasn1 python3-pyasn1-modules rpm-build \
+rsync rust systemd-devel uuid wget zlib-devel"
```

Comment:
I think it will look cleaner line by line because it makes it easier to see the list of dependencies and it's a cleaner git history if we edit any of the dependencies.

I think YAML allows the syntax like this:

```
run: >
  dnf -y install
  389-ds-base
  389-ds-base-snmp
  autoconf
  automake
  ...
```

Reply:
Perhaps it would be better if I put this command in a script? Then it can be reused for other jobs.

Reply:
Now I'm thinking, it'll be even better to use 'one source of truth' then...

We have dependencies in 'spec' file, so may be we can get them from there? Like we do in https://www.port389.org/docs/389ds/development/building.html

```
# git clone https://github.com/389ds/389-ds-base.git
# cd 389-ds-base
# sudo dnf install `grep "^BuildRequires" rpm/389-ds-base.spec.in | awk '{print $2}' | sed -e "s/%{python3_pkgversion}/3/"`
```

Reply:
I can't use this on Debian or SUSE. It's also out of sync from Fedora spec file (doesn't have `make` for example), it also pulls in `npm` and `nodejs` that are not needed for this test. So I'd prefer to maintain different lists of dependencies for each distro.